sys/luid/luid.c

```diff
@@ -37,10 +37,25 @@ void __attribute__((weak)) luid_base(void *buf, size_t len)
 #if CPUID_LEN
     uint8_t *out = (uint8_t *)buf;
     uint8_t cid[CPUID_LEN];
+    uint8_t sum = 0;
 
     cpuid_get(cid);
     for (size_t i = 0; i < MAX(len, CPUID_LEN); i++) {
-        out[i % len] ^= cid[i % CPUID_LEN];
+        uint8_t tmp, j, k;
+        j = i % len;
+        k = i % CPUID_LEN;
+
+        /* xor with CPU ID */
+        out[j] ^= cid[k];
+
+        /* get 'random' position */
+        sum += out[j];
+        k = sum % (j + 1);
+
+        /* shuffle bytes */
+        tmp    = out[j];
+        out[j] = out[k];
+        out[k] = tmp;
```

Comment:
Is this a common shuffling alorithm?

Reply:
It's basically [Fisher–Yates shuffle
](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle)

Reply:
Ah nice, mind adding a reference? You can squash right away!

Reply:
I added a comment.